qutebrowser/javascript/webelem.js

```diff
@@ -171,14 +203,72 @@ window._qutebrowser.webelem = (function() {
             }
         }
 
+        // Recurse into frames and add them
+        for (let i = 0; i < subelem_frames.length; i++) {
+            if (iframe_same_domain(subelem_frames[i])) {
+                const subelems = subelem_frames[i].document.
+                    querySelectorAll(selector);
+                const frame = subelem_frames[i];
+                for (let elem_num = 0; elem_num < subelems.length; ++elem_num) {
+                    if (!only_visible || is_visible(subelems[elem_num])) {
+                        out.push(serialize_elem(subelems[elem_num], frame));
+                    }
+                }
+            }
+        }
+
         return out;
     };
 
+    // Runs a function in a frame until the result is not null, then return
+    function run_frames(func) {
+        for (let i = 0; i < window.frames.length; ++i) {
+            if (iframe_same_domain(window.frames[i])) {
+                const frame = window.frames[i];
+                const result = func(frame);
+                if (result) {
+                    return result;
+                }
+            }
+        }
+        return null;
+    }
+
     funcs.find_id = (id) => {
         const elem = document.getElementById(id);
-        return serialize_elem(elem);
+        if (elem) {
+            return serialize_elem(elem);
+        }
+
+        const serialized_elem = run_frames((frame) => {
+            const element = frame.window.document.getElementById(id);
+            if (element) {
+                return serialize_elem(element, frame);
+            }
+            return null;
+        });
+
+        if (serialized_elem) {
+            return serialized_elem;
```

Comment:
This is `null` anyways when nothing was found, right? So you should be able to just `return serialized_elem` unconditionally here.